lib/spack/spack/provider_index.py

```diff
@@ -110,94 +196,20 @@ def update(self, spec):
-        return sorted(s.copy() for s in providers)
-
-    # TODO: this is pretty darned nasty, and inefficient, but there
-    # are not that many vdeps in most specs.
-    def _cross_provider_maps(self, lmap, rmap):
-        result = {}
-        for lspec, rspec in iproduct(lmap, rmap):
-            try:
-                constrained = lspec.constrained(rspec)
-            except spack.error.UnsatisfiableSpecError:
-                continue
-
-            # lp and rp are left and right provider specs.
-            for lp_spec, rp_spec in iproduct(lmap[lspec], rmap[rspec]):
-                if lp_spec.name == rp_spec.name:
-                    try:
-                        const = lp_spec.constrained(rp_spec, deps=False)
-                        result.setdefault(constrained, set()).add(const)
-                    except spack.error.UnsatisfiableSpecError:
-                        continue
-        return result
-
-    def __contains__(self, name):
-        """Whether a particular vpkg name is in the index."""
-        return name in self.providers
-
-    def satisfies(self, other):
-        """Check that providers of virtual specs are compatible."""
-        common = set(self.providers) & set(other.providers)
-        if not common:
-            return True
-
-        # This ensures that some provider in other COULD satisfy the
-        # vpkg constraints on self.
-        result = {}
-        for name in common:
-            crossed = self._cross_provider_maps(self.providers[name],
-                                                other.providers[name])
-            if crossed:
-                result[name] = crossed
-
-        return all(c in result for c in common)
-
     def to_json(self, stream=None):
+        """Dump a JSON representation of this object.
+
+        Args:
+            stream: stream where to dump
+        """
         provider_list = self._transform(
             lambda vpkg, pset: [
                 vpkg.to_node_dict(), [p.to_node_dict() for p in pset]], list)
 
         sjson.dump({'provider_index': {'providers': provider_list}}, stream)
 
-    @staticmethod
-    def from_json(stream):
-        data = sjson.load(stream)
-
-        if not isinstance(data, dict):
-            raise ProviderIndexError("JSON ProviderIndex data was not a dict.")
-
-        if 'provider_index' not in data:
-            raise ProviderIndexError(
-                "YAML ProviderIndex does not start with 'provider_index'")
-
-        index = ProviderIndex()
-        providers = data['provider_index']['providers']
-        index.providers = _transform(
-            providers,
-            lambda vpkg, plist: (
-                spack.spec.Spec.from_node_dict(vpkg),
-                set(spack.spec.Spec.from_node_dict(p) for p in plist)))
-        return index
-
     def merge(self, other):
-        """Merge `other` ProviderIndex into this one."""
+        """Merge another provider index into this one."""
```

Comment:
Since you changed the argument reference in the original docstring comment, do you want to add the standard `Args` definition to the comment?